src/popup/components/TransactionOverview.vue

```diff
@@ -5,106 +5,131 @@
+            title: root.$t('transaction.type.spendTx'),
           };
         case SCHEMA.TX_TYPE.contractCall: {
-          const direction = this.getTxDirection({ tx: this.tx });
+          const direction = getTxDirection.value({ tx: props.tx });
           const contract = {
-            address: this.tx.contractId,
-            url: this.getExplorerPath(this.tx.contractId),
-            label: this.$t(`transaction.overview.${this.isDexRecipient ? 'superheroDex' : 'contract'}`),
+            address: props.tx.contractId,
+            url: getExplorerPath.value(props.tx.contractId),
+            label: root.$t(`transaction.overview.${isDexRecipient.value ? 'superheroDex' : 'contract'}`),
           };
           return {
-            sender: direction === 'sent' ? this.account : contract,
-            recipient: direction === 'received' ? this.account : contract,
-            title: this.$t('transaction.type.contractCallTx'),
-            function: this.tx.function,
+            sender: direction === TX_FUNCTION_SENT ? parsedAccount.value : contract,
+            recipient: direction === TX_FUNCTION_RECEIVED ? parsedAccount.value : contract,
+            title: root.$t('transaction.type.contractCallTx'),
+            function: props.tx.function,
           };
         }
         case SCHEMA.TX_TYPE.contractCreate:
           return {
-            sender: this.account,
+            sender: parsedAccount.value,
             recipient: {
               contractCreate: true,
-              label: this.$t('transaction.overview.contractCreate'),
+              label: root.$t('transaction.overview.contractCreate'),
             },
-            title: this.$t('transaction.type.contractCreateTx'),
+            title: root.$t('transaction.type.contractCreateTx'),
           };
         case SCHEMA.TX_TYPE.namePreClaim:
         case SCHEMA.TX_TYPE.nameClaim:
         case SCHEMA.TX_TYPE.nameBid:
         case SCHEMA.TX_TYPE.nameUpdate:
           return {
-            sender: this.account,
+            sender: account.value,
             recipient: {
               aens: true,
-              label: this.$t('transaction.overview.aens'),
+              label: root.$t('transaction.overview.aens'),
             },
-            title: this.$t('transaction.type')[this.txType],
+            title: (root.$t('transaction.type') as any)[txType.value],
           };
         default:
           throw new Error('Unsupported transaction type');
       }
-    },
-    txType() {
-      return this.getTxType({ tx: this.tx });
-    },
-  },
-  async mounted() {
-    await watchUntilTruthy(() => this.$store.state.middleware);
-    if (this.tx.recipientId?.startsWith('nm_')) {
-      this.name = (await this.$store.state.middleware.getNameById(this.tx.recipientId)).name;
-    }
+    });
+
+    onMounted(async () => {
+      await watchUntilTruthy(() => middleware);
+      if (props.tx.recipientId?.startsWith('nm_')) {
```

Comment:
I suggest to use `validateHash` util function:
```
const { valid, prefix } = validateHash(props.tx.recipientId);
if (valid && prefix === HASH_PREFIX_NAME) {
  ...
```

(prefered) Or create separate util function similar to `isContract` that would wrap above logic.